src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -273,6 +274,25 @@ private static CSharpSyntaxContext CreateContextWorker(Document document, Semant
                 cancellationToken: cancellationToken);
         }
 
+        private static bool ComputeIsInTaskLikeTypeContext(SyntaxToken targetToken, SemanticModel semanticModel, ISet<SyntaxKind> precedingModifiers, CancellationToken cancellationToken)
+        {
+            // Check if we're immediately after 'async' (which the compiler either figured out was a keyword, or it
+            // thinks is an unbound identifier).
+            if (targetToken.Kind() == SyntaxKind.AsyncKeyword)
+                return true;
+
+            if (targetToken is { RawKind: (int)SyntaxKind.IdentifierToken, Text: "async" } &&
+                semanticModel.GetSymbolInfo(targetToken.GetRequiredParent(), cancellationToken).GetAnySymbol() is null)
+            {
+                return true;
```

Comment:
In my initial logic I only checked `precedingModifiers` to contain `SyntaxKind.AsyncKeyword`. Why have you changed it? `precedingModifiers` are smart enough and will contain `SyntaxKind.AsyncKeyword` even when `async` is an identifier, I checked it. I think you can just remove this method and use directly  `precedingModifiers.Contains(SyntaxKind.AsyncKeyword)` in the place, where this method is currently used, it will be pretty self-explanatory IMO.

Reply:
Looking!

Reply:
Thanks.  That worked! :)